File: etlAuto/etl/ControlFileCompare.py
# -*- coding: utf-8 -*-
import os
import sys
import time
import datetime
import logging
from ETL import ETL

logging.basicConfig(level=logging.INFO)

class ControlFileCompare:
    def compare(self,a,b):
        date1 = a[len(a)-8:len(a)]
        date2 = b[len(b)-8:len(b)]
        logging.info(date1+" and "+date2)
        result = cmp(date1,date2)
        return result

if __name__ == '__main__':
    a = 'xxx.20170819'
    b = 'xxx.20170820'
    compare = ControlFileCompare()
    print compare.compare(a,b)

File: etlAuto/etl/ControlFileCompare.java
package etl;

import java.util.Comparator;

class ControlFileCompare
  implements Comparator<String>
{
  public int compare(String a, String b)
  {
    String date1 = a.substring(a.length() - 8);
    String date2 = b.substring(b.length() - 8);
    return date1.compareTo(date2);
  }
}
